### DISCUSSION TITLE
working of insert_many vs multiple single inserts

### DISCUSSION THREAD

**Discussion by vikigenius** [SeaQL/sea-orm#453] [Q&A] [answered]
I want to know about the optimization of insert_many. How is it implemented? Does it work as a single insert with multiple values like SQLITE allows

```
INSERT INTO 'tablename' ('column1', 'column2') VALUES
  ('data1', 'data2'),
  ('data1', 'data2'),
  ('data1', 'data2'),
  ('data1', 'data2');
```

or does it work as multiple insert statements, which is presumably slower?

**Comment by billy1624** [comment, marked as answer]
Hey @vikigenius, just like what you have said. `insert_many` will construct SQL statement like that. And it will be sent to the database once only.

https://github.com/SeaQL/sea-orm/blob/6120d829245a3fb672d79c45f1860c06047921ec/src/entity/base_entity.rs#L384-L406
